Stripe/UIImage+Stripe.m

```diff
@@ -13,17 +13,32 @@ @implementation UIImage (Stripe)
 - (NSData *)stp_jpegDataWithMaxFileSize:(NSUInteger)maxBytes {
     CGFloat scale = 1.0;
     NSData *imageData = UIImageJPEGRepresentation(self, 0.5);
-    while (imageData.length > maxBytes) {
-        scale = scale * (CGFloat) 0.8;
-        CGSize newImageSize = CGSizeMake(self.size.width * scale,
-                                         self.size.height *scale);
-        UIGraphicsBeginImageContextWithOptions(newImageSize, NO, self.scale);
-        [self drawInRect:CGRectMake(0, 0, newImageSize.width, newImageSize.height)];
-        UIImage *newImage = UIGraphicsGetImageFromCurrentImageContext();
-        UIGraphicsEndImageContext();
-        imageData = UIImageJPEGRepresentation(newImage, 0.5);
-    }
 
+    // Try something smarter first
+    if (imageData.length > maxBytes) {
+        // Assuming jpeg file size roughly scales linearly with area of the image
+        // which is ~correct (although breaks down at really small file sizes)
+        CGFloat percentSmallerNeeded = (CGFloat) maxBytes / imageData.length;
+
+        // Shrink to a little bit less than we need to try to ensure we're under
+        // (otherwise its likely our first pass will be over the limit due to
+        // compression variance and floating point rounding)
+        scale = scale * sqrt(percentSmallerNeeded - (percentSmallerNeeded * 0.05));
```

Comment:
It's just area scaling plus a little extra. Eg if you have something that is 100x100 (area == 10k) and you want to make it area == 5k, (make it 50% of the original size) you have to scale down the height and width by sqrt(0.5) ~= 0.71 (100*0.71 ~= 71, 71*71 == 5041). 

To try to ensure it actually goes under the max byte limit after rounding and re-jpeging (since the jpeg algo isn't actually 100% linear on area) I just scaled it down a little extra than needed (and extra 5% * whatever we were scaling it by). Not sure that's the best way to do "a little extra", could try some other things if you have suggestions.